examples/shopping-cart/src/main/kotlin/com/example/shoppingcart/ShoppingCartEntity.kt

```diff
@@ -1,25 +1,26 @@
-package com.example.shoppingcart
-
+// #example-shopping-cart-kotlin
+import com.example.shoppingcart.Shoppingcart
 import com.example.shoppingcart.persistence.Domain
 import com.google.protobuf.Empty
 import io.cloudstate.javasupport.EntityId
-import io.cloudstate.javasupport.eventsourced.*
+import io.cloudstate.javasupport.eventsourced.CommandContext
+import io.cloudstate.kotlinsupport.api.eventsourced.*
 
 import java.util.stream.Collectors
 
 @EventSourcedEntity
 class ShoppingCartEntity(@param:EntityId private val entityId: String) {
-    private val cart: MutableMap<String, Shoppingcart.LineItem?> = LinkedHashMap()
+    private val cart: MutableMap<String, Shoppingcart.LineItem?> = mutableMapOf<String, Shoppingcart.LineItem?>()
```

Comment:
Does this preserve order?

Reply:
Yes!
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/mutable-map-of.html